db/fixtures/ae_datastore/ManageIQ/Deployment/ContainerProvider/System/StateMachine/Deployment.class/__methods__/post_provision.rb

```diff
@@ -1 +1,60 @@
+def find_vm_by_tag(tag)
+  tag = "/managed/deploy/#{tag}_#{$evm.root['automation_task'][:id]}"
+  $evm.vmdb(:vm).find_tagged_with(:any => tag, :ns => "*")
+end
 
+def assign_vms_to_deployment_nodes(masters, nodes)
+  deployment = $evm.vmdb(:container_deployment).find(
+    $evm.root['automation_task'].automation_request.options[:attrs][:deployment_id])
+  masters.each do |master|
+    deployment.assign_container_deployment_node(master.id, "master")
+  end
+  nodes.each do |node|
+    deployment.assign_container_deployment_node(node.id, "node")
+  end
+  $evm.root['deployment_master'] = deployment.roles_addresses("deployment_master")
+  $evm.root['inventory'] = deployment.regenerate_ansible_inventory
+  $evm.root['rhel_subscribe_inventory'] =  deployment.regenerate_ansible_subscription_inventory
+end
+
+def missing_subscription_fields?
+  $evm.root['rhsub_user'].nil? || $evm.root['rhsub_pass'].nil? || $evm.root['rhsub_sku'].nil?
+end
+
+def extract_ips(vms)
+  ips = []
+  vms.each do |vm|
+    ips << vm.hardware.ipaddresses.last unless vm.hardware.ipaddresses.empty?
```

Comment:
`unless vm.hardware.ipaddresses.empty?`
@alongoldboim does that covers the problem we saw in AWS with missing ip?